src/admin/redux/modules/email/reducer.js

```diff
@@ -81,8 +82,14 @@ const template_subs = ifElse
 
 const indexByProp = compose(indexBy, propOr(''))
 
+const greeting = member => assoc
+  ( 'greeting'
+  , member.first_name || member.title + ' ' + member.last_name
+  , member
+  )
+
 const [ primaries, secondaries ] =
-  map(indexByProp, [ 'primary_email', 'secondary_email' ])
+  map(compose(map(map(greeting)), indexByProp), [ 'primary_email', 'secondary_email' ])
```

Comment:
The `map(compose(map(map` could also be written `map(map(map(map` or `map(compose(map(compose`. It goes over each element of `[ 'primary_email', 'secondary_email' ]` (the first map), then applies `indexByProp`, returning a function. This is fed into `map(map(greeting))` (it's fed into it by `compose`).

So at this point we have things that look like `map(map(greeting), indexByProp('primary_email'))`. So let's call `map(greeting)` `f` and `indexByProp('primary_email')` `g`. So we have `map(f, g)`. Huh? Isn't map for arrays? Or maybe objects?

Well it turns out *functions* have a functor instance! And for them, `map` is just composition. So here we are composing together `map(greeting)` and `indexByProp('primary_email')`.

So putting it altogether, once we get an array of member objects, we'll index them by, say, `'primary_email'`, then apply `greeting` to each member.

Does that help at all?